docs/getting_started.rst

```diff
@@ -1,135 +1,288 @@
+            },
+            {
+                "data": {
+                    "core-table": pd.DataFrame({"col1": ["y"]}),
+                    "aux-table": pd.DataFrame({"f": [3.2]}),
+                }
+            },
+        ]
+
+        dm = store_dataframes_as_dataset(store_factory, dataset_uuid="two-tables", dfs=dfs)
+        dm.tables
+        sorted(dm.partitions.keys())
 
-   df = read_table("a_unique_dataset_identifier", store, table="table")
-   df
 
-To understand the basics of the dataset, we can look at the files that were
-written using the store method. The main file of a dataset is
-``<dataset_uuid>.by-dataset-metadata.json``. Here we track all partitions that
-exist inside a datasets as well as the tables and additional metadata. This
-file is loaded on any operation of the dataset.
+Reading data from storage
+=========================
 
-The magic ``<dataset_uuid>/<table>/_common_metadata`` file is an Apache Parquet
-file that contains no data. It is simply used to persist the schema of a single
-table.
+After we have written the data, we may want to read it back in again. For this we can
+use :func:`~kartothek.io.eager.read_table`. This method returns the complete
+table of the dataset as a pandas DataFrame.
+
+.. ipython:: python
+
+    from kartothek.io.eager import read_table
+
+    read_table("a_unique_dataset_identifier", store_factory, table="table")
 
-Finally ``<dataset_uuid>/<table>/<partition_label>.parquet`` is the file that
-contains the data for this partition in the specific table. By default
-``kartothek`` serializes data to Apache Parquet files but also supports other
-file formats like CSV.
+
+We can also read a dataframe iteratively, using 
+:func:`~kartothek.io.iter.read_dataset_as_dataframes__iterator`. This will return a generator
+of dictionaries (one dictionary for each `partition`), where the keys of each dictionary
+represent the `tables` of the dataset. For example,
 
 .. ipython:: python
 
-   list(store.keys())
+    from kartothek.io.iter import read_dataset_as_dataframes__iterator
+
+    for partition_index, df_dict in enumerate(
+        read_dataset_as_dataframes__iterator(dataset_uuid="two-tables", store=store_factory)
+    ):
+        print(f"Partition #{partition_index}")
+        for table_name, table_df in df_dict.items():
+            print(f"Table: {table_name}. Data: \n{table_df}")
+
+
+.. admonition:: Filtering the dataset using predicates
+
+    It is possible to filter data during reads using simple predicates by using
+    the ``predicates`` argument. Technically speaking, ``kartothek`` supports predicates
+    in `disjunctive normal form <https://en.wikipedia.org/wiki/Disjunctive_normal_form>`_.
+
+    When this argument is defined, ``kartothek`` tries to use the Apache Parquet metadata
+    for partition pruning and predicate pushdown to speed up queries when possible. How this
+    works is a complex topic, well beyond the scope of this guide but it is a 'good-to-know'
```

Comment:
I'd scratch the "good-to-know" part and simply refer the user to the appropriate section (Create one if necessary and add a TODO)